### PR TITLE
DT-916 Change contract for replace NOMS number - when existing offend…

### PR DESCRIPTION
**Pull request by andymarke**
…er found return  400 rather then fail silently

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/CustodyResource.java [hunk 1]
```diff
@@ -86,7 +86,7 @@ public IDs updateOffenderNomsNumber(final @PathVariable String crn,
 
     @RequestMapping(value = "offenders/nomsNumber/{originalNomsNumber}/nomsNumber", method = RequestMethod.PUT, consumes = "application/json")
     @ApiResponses(value = {
-            @ApiResponse(code = 400, message = "The new noms number is not present"),
+            @ApiResponse(code = 400, message = "The new noms number is not present in request or existing offender exists with then new NOMS number already"),
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/CustodyResource.java:89]: could possibly model as a `409 Conflict`?

Comment (andymarke) [reply src/main/java/uk/gov/justice/digital/delius/controller/secure/CustodyResource.java:89]: @andrewrlee I like that a lot. @mikehalmamoj what do you think?

Comment (mikehalmamoj) [reply src/main/java/uk/gov/justice/digital/delius/controller/secure/CustodyResource.java:89]: :+1:

File: src/main/java/uk/gov/justice/digital/delius/service/OffenderIdentifierService.java [hunk 2]
```diff
@@ -57,7 +58,7 @@ public IDs updateNomsNumber(String crn, UpdateOffenderNomsNumber updateOffenderN
 
         if (!existingOffendersAlreadyWithNomsNumber.isEmpty()) {
             log.info("No need to Update NOMS number since an offender with the new noms number already exists");
-            return existingOffendersAlreadyWithNomsNumber.stream().map(OffenderTransformer::idsOf).collect(Collectors.toList());
+            throw new ConflictingRequestException(String.format("NOMS number %s is already assigned to %s", updateOffenderNomsNumber.getNomsNumber(), existingOffendersAlreadyWithNomsNumber.stream().map(Offender::getCrn).collect(Collectors.joining(","))));
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/service/OffenderIdentifierService.java:61]: Slightly archaic control by exception, but this follows existing pattern.  Something that we may want to move to Result type responses in the future ...

**Pull request merged**